src/browser/BrowserService.cpp

```diff
@@ -219,6 +219,35 @@ QJsonObject BrowserService::getDatabaseGroups()
     return result;
 }
 
+QJsonArray BrowserService::getDatabaseEntries()
+{
+    auto db = getDatabase();
+    if (!db) {
+        return {};
+    }
+
+    Group* rootGroup = db->rootGroup();
+    if (!rootGroup) {
+        return {};
+    }
+
+    QJsonArray entries;
+    for (const auto& group : rootGroup->groupsRecursive(true)) {
+        if (group->isRecycled()) {
```

Comment:
You should just exclude the recycle bin group itself, then anything under it will be excluded (groups and entries) `group == db->metadata()->recycleBin()`

Reply:
Done. Part of aff8519 Add 'get-database-entries' Proxy Request